data-integration/utilities.py

```diff
@@ -165,18 +163,18 @@ def datasetExists(
     ds_id : int.
     dbURL : str, optional
         The default is "postgresql://test:example@localhost:5433/dataset".
-    tables : list of strings, optional
-        The default is ["datasets","spatial","data"].
     Returns
     -------
     bool
     """
+    tables = ["datasets", "spatial", "data"]
     engine = sqla.create_engine(dbURL)
     lengths = []
     for table in tables:
         with engine.connect() as con:
             rs = con.execute(
-                "SELECT COUNT(*) FROM {} WHERE ds_id = {}".format(table, ds_id)
+                "SELECT COUNT(*) FROM {} WHERE ds_id = %(ds_id)s;".format(table),
+                {"ds_id": ds_id},
             )
             for row in rs:
```

Comment:
Using a for loop is useless in this case since the query gives only one line result, and will lead to bugs if the query is changed by one giving several line results.

Reply:
The query gives more than one result, that's why a for loop is used. But I agree that is not elegant at the moment. I am open to suggestions in case. Anyway, I am trying to rebuild this function using psycopg2 (see thread with Malik), so this might be disappear as well.

Reply:
You are right, the description is not clear. I did want an `or` operator here.

Reply:
I cleaned this a bit in the new commit (which uses psycopg2 directly).

Reply:
It looks better :)
count(*) will return a single line, you can use fetchone() to get the result. What do you think of the following:

```python
cursor = con.cursor()
cursor = cursor.execute("SELECT COUNT(*) FROM table WHERE ds_id = %(ds_id)s;", {'ds': ds})
result = cursor.fetchone()
if result:
  match_count = result[0]
```